yunwoo/baekjoon/브루트포스/2798-블랙잭.js

```diff
@@ -0,0 +1,39 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [n, m] = input[0].split(" ").map(Number);
+  const positiveInteger = input[1]
+    .split(" ")
+    .map(Number)
+    .sort((a, b) => b - a);
+  let result = 0;
+
+  for (let i = 0; i < n - 2; i++) {
+    for (let j = i + 1; j < n - 1; j++) {
+      for (let k = j + 1; k < n; k++) {
+        let sum = positiveInteger[i] + positiveInteger[j] + positiveInteger[k];
+        if (sum > result && sum <= m) {
+          result = sum;
+        }
+      }
+    }
+  }
+
+  console.log(result);
+});
```

Comment:
```suggestion
  // 풀이
  const [n, m] = input[0].split(" ").map(Number);
  const positiveInteger = input[1].split(" ").map(Number);
  let result = 0;

  for (let i = 0; i < n - 2; i++) {
    for (let j = i + 1; j < n - 1; j++) {
      for (let k = j + 1; k < n; k++) {
        const sum = positiveInteger[i] + positiveInteger[j] + positiveInteger[k];
        
        if (sum <= m) {
          result = Math.max(result, sum);
        }
      }
    }
  }

  console.log(result);
});
```
- 풀이 잘봤습니다 🙇‍♂️
- 해당 풀이는 정렬과 관계없이 3중 for문으로 모두 탐색을 해야되는 브루트포스(완전탐색)문제라서 sort는 제거해주었습니다.
- 3번째 for문에서 sum은 const로 선언해주세요!
```js
if (sum > result && sum <= m) {
  result = sum;
}
```
- 위 코드는 Math.max를 이용한 아래와 같은 방법도 있습니다.
```js
if (sum <= m) {
  result = Math.max(result, sum);
}
```

Reply:
- 오 Math.max를 이용하면 조건식을 깔끔하게 가져갈 수 있네요!!
- sort가 필요가 없었군요 ㅎㅎ 브루트포스(완전탐색)에 대해 알아봐야겠네요 👍
- let 실수 주의하겠습니다 감사합니다 🙇‍♂️